common/meta/types/src/meta_storage_errors.rs

```diff
@@ -0,0 +1,161 @@
+// Copyright 2021 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fmt::Display;
+
+use common_exception::ErrorCode;
+use serde::Deserialize;
+use serde::Serialize;
+use sled::Error as SledError;
+use thiserror::Error;
+
+use crate::MetaError;
+
+#[derive(Error, Serialize, Deserialize, Debug, Clone, PartialEq)]
+pub enum MetaStorageError {
+    // type to represent serialize errors
+    #[error("{0}")]
+    SerializeError(String),
+
+    #[error("{0}")]
+    BadBytes(String),
+
+    #[error("{0}")]
+    StorageError(String),
+
+    #[error("{0}")]
+    MetaStoreDamaged(String),
+
+    #[error("{0}")]
+    TransactionAbort(String),
+
+    #[error("{0}")]
+    TransactionError(String),
+
+    #[error("{0}")]
+    MetaSrvError(String),
+
+    #[error("{0}")]
+    UnknownDatabase(String),
+
+    #[error("{0}")]
+    UnknownDatabaseId(String),
+}
+
+pub type MetaStorageResult<T> = std::result::Result<T, MetaStorageError>;
+
+impl From<MetaError> for MetaStorageError {
+    fn from(e: MetaError) -> Self {
+        MetaStorageError::MetaSrvError(format!("metasrv error: {:?}", e))
+    }
+}
```

Comment:
I suppose MetaError is a superset of MetaStorageError.
If it is, it should not need to convert MetaError ToMetaStorageError.